projects/packages/sync/src/sync-queue/class-queue-storage-table.php

```diff
@@ -456,4 +476,190 @@ public function delete_items_by_ids( $ids ) {
 			)
 		);
 	}
+
+	/**
+	 * Table initialization
+	 */
+	public static function initialize_custom_sync_table() {
+		/**
+		 * Initialize an instance of the class with a test name, so we can use table prefix and then test if the table is healthy.
+		 */
+		$custom_table_instance = new Queue_Storage_Table( 'test_queue' );
+
+		// Check if the table exists
+		if ( ! $custom_table_instance->custom_table_exists() ) {
+			$custom_table_instance->create_table();
+		}
+
+		if ( ! $custom_table_instance->is_custom_table_healthy() ) {
+
+			// TODO: clean up the table.
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Migrates the existing Sync events from the options table to the Custom table
+	 *
+	 * @return void
+	 */
+	public static function migrate_from_options_table_to_custom_table() {
+		global $wpdb;
```

Comment:
Do you think we should delete any leftover rows from the custom table first before populating it from the options?

Reply:
I am leaving that logic for the follow-up PR, as we discussed earlier today - to do a two-step migration to avoid any leftover data from parallel requests being stuck in the table.

If we clean up the table in the migration code, we'll delete valid data and the clean up should happen outside of this method.